pkg/openshift/utils.go

```diff
@@ -0,0 +1,29 @@
+package openshift
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+func NamespaceComponent(applicationName string, componentName string) (string, error) {
+
+	// Error if it's blank
+	if applicationName == "" {
+		return "", errors.New("namespacing: application name cannot be blank")
+	}
+
+	// Error if it's blank
+	if componentName == "" {
+		return "", errors.New("namespacing: component name cannot be blank")
+	}
+
+	// Some parts of the code have redundant / recursive functions that will call NamespaceOpenShiftComponent twice.
```

Comment:
We should probably fix it there, rather than doing it like this.

If this is happening, that it is an indication that we are doing something wrong there.

Reply:
I tried troubleshooting but gave up, I'll find the issue and correct it.